lte/gateway/c/session_manager/SetMessageManagerHandler.cpp

```diff
@@ -196,4 +247,99 @@ void SetMessageManagerHandler::initiate_release_session(
+              << " of session Id " << session->get_session_id();
+  if (notif->common_context().sm_session_version() == 0) {
+    Status status(grpc::OUT_OF_RANGE, "Version number Out of Range");
+    response_callback(status, SmContextVoid());
+    return;
+  }
+  if (notif->rat_specific_notification().request_type() !=
+      EXISTING_PDU_SESSION) {
+    MLOG(MINFO) << " Wrong request type for sesion id"
+                << session->get_session_id();
+    Status status(grpc::UNKNOWN, "Unknown session request type");
+    response_callback(status, SmContextVoid());
+    return;
+  }
+  auto session_update = SessionStore::get_default_session_update(session_map);
+  auto session_id     = session->get_session_id();
+  SessionStateUpdateCriteria& session_crit = session_update[imsi][session_id];
+  m5g_enforcer_->m5g_state_change_action(
+		  imsi, session_crit, *notif, **session_it);
+  bool update_success = session_store_.update_sessions(session_update);
+  if (update_success) {
+     MLOG(MINFO) << "Successfully updated SessionStore "
+		  << "of subscriber" << imsi;
+     response_callback(Status::OK, SmContextVoid());
+     return;
+  }
+  Status status(grpc::ABORTED, "update operation aborted in  middle");
+  response_callback(status, SmContextVoid());
+  return;
+}
+void SetMessageManagerHandler::idle_mode_change_sessions_handle(
+    const SetSmNotificationContext* notif,
+    std::function<void(Status, SmContextVoid)> response_callback) {
+  // extract IMSI value from proto
+  auto imsi = notif->common_context().sid().id();
+  if (notif->rat_specific_notification().request_type() !=
+      EXISTING_PDU_SESSION) {
+    MLOG(MINFO) << " Wrong request type for imsi" << imsi;
+    Status status(grpc::UNKNOWN, "Unknown session request type");
+    response_callback(status, SmContextVoid());
+    return;
+  }
+  if (notif->common_context().sm_session_version() == 0) {
+    Status status(grpc::OUT_OF_RANGE, "Version number Out of Range");
+    response_callback(status, SmContextVoid());
+    return;
+  }
+  auto session_map = session_store_.read_sessions({imsi});
+  int count        = 0;
+  for (auto& session : session_map[imsi]) {
+    auto session_update = SessionStore::get_default_session_update(session_map);
+    auto session_id     = session->get_session_id();
+    SessionStateUpdateCriteria& session_crit = session_update[imsi][session_id];
+    m5g_enforcer_->m5g_state_change_action(imsi, session_crit, *notif, session);
+    bool update_success = session_store_.update_sessions(session_update);
+    if (update_success) {
+      MLOG(MINFO) << "Successfully updated SessionStore "
+                  << " session id: " << session->get_session_id() << imsi
+                  << " of imsi";
+      count++;
+      continue;
+    }
+    MLOG(MINFO) << "Operation aborted in  middle"
+                << " session id: " << session->get_session_id() << imsi
+                << " of imsi";
+  }
+  if (!count) {

```

Comment:
Right now this error message can be triggered either by session not found or all `update_sessions` fail.

Better you use .size to first see if no sessions were fond (as commented before) and then use this count to see if all the updates failed. 

What would be the difference on all updates failed and few updates failed?

Reply:
In few updates failed case,  only for the failed ones  "Operation aborted in middle" message comes.
 For all failed case, we c  multiple message plus "No valid session found"
 I slightly added code flag, that makes "count=0" log skipped when any of session is aborted.